### PR TITLE
Revert "prepared plan cache: Enable by default (#4036)"

### PR DESCRIPTION
**Pull request by eurekaka**
This reverts commit 6000b9584ce20466d64b4da1b7112e21b4284678.

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):
   https://github.com/pingcap/docs/pull/4036, previous docs PR link
   https://github.com/pingcap/tidb/pull/23240, related tidb PR link

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [x] Need modification after applied to another branch 
- [x] Might cause conflicts after applied to another branch

### CODE REVIEW

File: sql-prepare-plan-cache.md [hunk 1]
```diff
@@ -5,12 +5,14 @@ summary: Learn about SQL Prepare Execution Plan Cache in TiDB.
...
 
-- Using the `COM_STMT_PREPARE` and `COM_STMT_EXECUTE` protocol features.
-- Using the SQL statements `PREPARE` and `EXECUTE`.
+There are two forms of `Prepare` / `Execute` queries:
 
-The TiDB optimizer handles these two types of queries in the same way: when preparing, the parameterized query is parsed into an AST (Abstract Syntax Tree) and cached; in later execution, the execution plan is generated based on the stored AST and specific parameter values.
+- In the binary communication protocol, use `COM_STMT_PREPARE` and `COM_STMT_EXECUTE` to execute general parameterized SQL queries;
+- In the text communication protocol, use `COM_QUERY` to execute `Prepare` and `Execution` SQL queries.
+
+The optimizer handles these two types of queries in the same way: when preparing, the parameterized query is parsed into an AST (Abstract Syntax Tree) and cached; in later execution, the execution plan is generated based on the stored AST and specific parameter values.
```
Comment (morgo) [inline sql-prepare-plan-cache.md:15]: I think it would be nice to keep this text (it reads a bit clearer). But the rest LGTM

Comment (eurekaka) [reply sql-prepare-plan-cache.md]: Updated, PTAL

**Comment by morgo** [comment]
LGTM

**Comment by morgo** [comment]
It looks like this needs a 5.0 cherry pick as well. I'm not sure why my bot command failed, but I don't have permissions through the GitHub GUI to add the labels.

File: sql-prepare-plan-cache.md [hunk 2]
```diff
@@ -5,7 +5,7 @@ summary: Learn about SQL Prepare Execution Plan Cache in TiDB.
 
 # SQL Prepare Execution Plan Cache
 
-In the current development release of TiDB, the execution plan of prepared statements is cached by default. This includes both forms of prepared statements:
+TiDB supports execution plan caching for `Prepare` / `Execute` queries. This includes both forms of prepared statements:
```
Comment (TomShawn) [inline sql-prepare-plan-cache.md:8]:
```suggestion
TiDB supports execution plan caching for `Prepare` and `Execute` queries. This includes both forms of prepared statements:
```

File: sql-prepare-plan-cache.md [hunk 3]
```diff
@@ -44,7 +44,7 @@ There are two points worth noting about execution plan caching and query perform
...
 
-The default capacity for the plan cache is `100` statements. You can configure this by modifying [`prepare-plan-cache`](/tidb-configuration-file.md#prepared-plan-cache) in the TiDB configuration file.
+Currently, the execution plan cache is disabled by default. You can enable this feature by enabling the [`prepare-plan-cache`](/tidb-configuration-file.md#prepared-plan-cache) in the configuration file.
```
Comment (TomShawn) [inline sql-prepare-plan-cache.md:47]: Why deleting `TiDB` here?

Comment (eurekaka) [reply sql-prepare-plan-cache.md:47]: It is a revert commit.

File: sql-prepare-plan-cache.md (hunk 2 above)
Comment (eurekaka) [reply sql-prepare-plan-cache.md:8]: The plan cache is not for "Prepare and Execute", it is for "Prepare" / "Execute" combination.

File: sql-prepare-plan-cache.md (hunk 3 above)
Comment (TomShawn) [reply sql-prepare-plan-cache.md:47]: There are also TiKV and PD configuration files. Suggest clearly specifying "TiDB configuration file".

File: sql-prepare-plan-cache.md (hunk 2 above)
Comment (TomShawn) [reply sql-prepare-plan-cache.md:8]: Does "/" means AND or OR here?

Comment (eurekaka) [reply sql-prepare-plan-cache.md:8]: Users would first use `Prepare` for a query, then use several consecutive `Execute` for the prepared statement. 

The feature is to cache the plan, while `Prepare` does not generate a plan actually, the plan is built in the first `Execute` and cached then, so strictly we cannot say "TiDB supports execution plan caching for `Prepare` queries".

Comment (TomShawn) [reply sql-prepare-plan-cache.md]: Shall we remove `Prepare` here? "TiDB supports execution plan caching for `Execute` queries."?

Comment (eurekaka) [reply sql-prepare-plan-cache.md]: The feature was called "prepare-plan-cache" historically...

Comment (eurekaka) [reply sql-prepare-plan-cache.md]: Changed to "`Prepare` and `Execute`" now.

**Comment by TomShawn** [comment]
@eurekaka Please resolve the conflict. Rest LGTM

**Comment by TomShawn** [comment]
> LGTM

+ If you find your command fails, it might be that the GitHub webhook is unstable. Please try again.
+ Since the Ti Chi bot is introduced, `LGTM` is an invalid command. You need to use `/lgtm` to approve a PR.

**Comment by TomShawn** [comment]
@eurekaka Is this PR v5.0 GA related?

**Comment by eurekaka** [comment]
@tomshawn ptal

**Comment by TomShawn** [comment]
@eurekaka Because this is a 5.0 GA-related content, its merge is planned and controlled. We'll merge this PR later.

**Pull request merged**